File: scripts/optimize-images.py
#!/usr/bin/env python3
"""Optimize intro-bg.png image for faster loading"""

import os
from PIL import Image

# Define paths - absolute path to the image
intro_bg_path = '/vercel/share/v0-project/public/images/intro-bg.png'
images_dir = os.path.dirname(intro_bg_path)
intro_bg_webp = os.path.join(images_dir, 'intro-bg.webp')

if not os.path.exists(intro_bg_path):
    print(f"[v0] Error: {intro_bg_path} not found")
    exit(1)

# Get original file size
original_size = os.path.getsize(intro_bg_path)
print(f"[v0] Original intro-bg.png size: {original_size / 1024:.2f} KB")

try:
    # Open and optimize PNG
    img = Image.open(intro_bg_path)
    print(f"[v0] Image size: {img.size}, Mode: {img.mode}")
    
    # Resize if too large (max width 1920 for desktop)
    if img.width > 1920:
        ratio = 1920 / img.width
        new_size = (1920, int(img.height * ratio))
        img = img.resize(new_size, Image.Resampling.LANCZOS)
        print(f"[v0] Resized to: {img.size}")
    
    # Save optimized PNG
    img.save(intro_bg_path, 'PNG', optimize=True, quality=85)
    optimized_png_size = os.path.getsize(intro_bg_path)
    print(f"[v0] Optimized PNG size: {optimized_png_size / 1024:.2f} KB (reduction: {(1 - optimized_png_size / original_size) * 100:.1f}%)")
    
    # Save as WebP for better compression
    img.save(intro_bg_webp, 'WEBP', quality=85)
    webp_size = os.path.getsize(intro_bg_webp)
    print(f"[v0] WebP version created: {webp_size / 1024:.2f} KB (reduction: {(1 - webp_size / original_size) * 100:.1f}%)")
    
    print("[v0] Image optimization complete!")
    print("[v0] Update start-screen.tsx to use WebP: backgroundImage: 'url(/images/intro-bg.webp)'")
    
except Exception as e:
    print(f"[v0] Error optimizing image: {e}")
    exit(1)

File: scripts/optimize-images.mjs
import sharp from 'sharp';
import fs from 'fs';
import path from 'path';

const imagePath = path.join(process.cwd(), 'public/images/intro-bg.png');
const outputPath = path.join(process.cwd(), 'public/images/intro-bg-optimized.png');

// Also create WebP version for better compression
const webpPath = path.join(process.cwd(), 'public/images/intro-bg.webp');

async function optimizeImage() {
    try {
        if (!fs.existsSync(imagePath)) {
            console.log('[v0] Image not found at:', imagePath);
            return;
        }

        console.log('[v0] Optimizing intro-bg.png...');
        
        // Get original file size
        const originalSize = fs.statSync(imagePath).size;
        console.log('[v0] Original size:', (originalSize / 1024).toFixed(2), 'KB');

        // Optimize PNG: reduce resolution, quality, and use compression
        const pngBuffer = await sharp(imagePath)
            .resize(1920, 1080, {
                fit: 'cover',
                withoutEnlargement: true
            })
            .png({ quality: 80, progressive: true, compressionLevel: 9 })
            .toBuffer();

        fs.writeFileSync(imagePath, pngBuffer);
        const newSize = fs.statSync(imagePath).size;
        const savings = ((1 - newSize / originalSize) * 100).toFixed(1);
        console.log('[v0] Optimized PNG size:', (newSize / 1024).toFixed(2), 'KB');
        console.log('[v0] Size reduction:', savings + '%');

        // Create WebP version for better compression (if supported by browser)
        const webpBuffer = await sharp(imagePath)
            .resize(1920, 1080, {
                fit: 'cover',
                withoutEnlargement: true
            })
            .webp({ quality: 75 })
            .toBuffer();

        fs.writeFileSync(webpPath, webpBuffer);
        console.log('[v0] Created WebP version:', (webpBuffer.length / 1024).toFixed(2), 'KB');
        console.log('[v0] Image optimization complete!');
    } catch (error) {
        console.error('[v0] Error optimizing image:', error.message);
    }
}

optimizeImage();
